src/cryptography/hazmat/bindings/openssl/binding.py

```diff
@@ -140,6 +140,21 @@ def _ensure_ffi_initialized(cls):
                 # adds all ciphers/digests for EVP
                 cls.lib.OpenSSL_add_all_algorithms()
                 cls._register_osrandom_engine()
+                # As of OpenSSL 3.0.0 we must register a legacy cipher provider
+                # to get RC2 (needed for junk asymmetric private key crypto),
+                # RC4, Blowfish, IDEA, SEED, etc. These things are ugly legacy,
+                # but we aren't going to get rid of them any time soon.
+                if cls.lib.CRYPTOGRAPHY_OPENSSL_300_OR_GREATER:
+                    legacy = cls.lib.OSSL_PROVIDER_load(
+                        cls.ffi.NULL, b"legacy"
+                    )
+                    _openssl_assert(cls.lib, legacy != cls.ffi.NULL)
+                    default = cls.lib.OSSL_PROVIDER_load(
+                        cls.ffi.NULL, b"default"
+                    )
+                    _openssl_assert(cls.lib, default != cls.ffi.NULL)
+                    # Store the references just in case we need them later
+                    cls.lib._providers = {"legacy": legacy, "default": default}
```

Comment:
Make this a list or something, not a dict.

Reply:
If we make it a list we'll have to bind `OSSL_PROVIDER_name` to know which provider is which. What makes a dict a bad choice here in your mind?

Reply:
We're not doing anything with these values, they're purely a keep-alive (and not even one that's required at that).

Reply:
That sounds like an argument to not store them at all more than keeping a list.